python/dgl/transform.py

```diff
@@ -383,6 +384,137 @@ def reverse(g, share_ndata=False, share_edata=False):
         g_reversed._edge_frame = g._edge_frame
     return g_reversed
 
+def reverse_heterograph(g, share_ndata=True, share_edata=False):
+    r"""Return the reverse of a graph
+
+    The reverse (also called converse, transpose) of a graph with edges
+    :math:`(i_1, j_1), (i_2, j_2), \cdots` is a new graph with edges
+    :math:`(j_1, i_1), (j_2, i_2), \cdots`.
+
+    For a heterograph with multiple edge types, we can treat edges corresponding to each
+    type as a separate graph and compute the reverse for each of them.
+
+    Given a :class:`DGLGraph` object, we return another :class:`DGLGraph` object
+    representing its reverse.
+
+    Notes
+    -----
+    * We do not dynamically update the topology of a graph once that of its reverse changes.
+      This can be particularly problematic when the node/edge attrs are shared. For example,
+      if the topology of both the original graph and its reverse get changed independently,
+      you can get a mismatched node/edge feature.
```

Comment:
I think this is no longer an issue since we change from sharing frames to sharing tensors. We can use the following sentences for the nodes instead.

    * If ``share_ndata`` or ``share_edata`` is ``True``, same tensors will be used for 
      the features of the original graph and the reversed graph. As a result, users 
      should avoid performing in-place operations on the features of the reversed 
      graph, which will corrupt the features of the original graph as well. For 
      concrete examples, refer to the ``Examples`` section below.